packages/fxa-auth-server/scripts/move-customers-to-new-plan-v2/move-customers-to-new-plan-v2.ts

```diff
@@ -0,0 +1,273 @@
+    );
+
+    if (!invoice.paid) {
+      throw new Error("Customer is pending renewal right now!");
+    }
+
+    const billedAt = new Date(invoice.created * 1000);
+    const oneDayMs = 1000 * 60 * 60 * 24;
+    const timeSinceBillMs = new Date().getTime() - billedAt.getTime();
+    const daysSinceBill = Math.floor(timeSinceBillMs / oneDayMs);
+    const refundAmount = daysSinceBill * this.proratedRefundRate;
+
+    if (refundAmount > invoice.amount_paid) {
+      throw new Error(`Will not refund ${invoice.id} for ${refundAmount} as it would eclipse the amount paid on the invoice`);
+    }
+
+    if (invoice.paid_out_of_band) {
+      const behavior = refundAmount === invoice.amount_paid ? {
+        refundType: RefundType.Full
+      } as const : {
+        refundType: RefundType.Partial,
+        amount: refundAmount,
+      } as const;
+
+      console.log(`Issuing ${refundAmount} (${behavior.refundType}) Paypal refund for ${invoice.id}`);
+      if (this.dryRun) {
+        console.log('(dry run mode, no refund issued)');
+      } else {
+        await this.paypalHelper.refundInvoice(invoice, behavior);
+      }
+    } else {
+      const chargeId =
+        typeof invoice.charge === 'string' ? invoice.charge : invoice.charge?.id;
+      if (!chargeId) {
+        throw new Error(`No charge for ${invoice.id}`);
+      }
+
+      console.log(`Issuing ${refundAmount} Stripe refund for ${invoice.id}`);
+      if (this.dryRun) {
+        console.log('(dry run mode, no refund issued)');
+      } else {
+        await this.enqueueRequest(() =>
+          this.stripe.refunds.create({
+            charge: chargeId,
+            amount: refundAmount
+          })
+        );
+      }
+    }
+
+    return refundAmount;
+  }
+
+  /**
+   * Check if a customer's list of subscriptions contains an excluded price ID
+   * @param subscriptions List of subscriptions to check for an excluded price ID in
+   */
+  isCustomerExcluded(subscriptions: Stripe.Subscription[]) {
+    for (const subscription of subscriptions) {
+      for (const item of subscription.items.data) {
+        if (this.excludeCustomersHavingPriceIds.includes(item.plan.id)) return true;
+      }
+    }
+
+    return false;
+  }
+
+  async writeReportHeader() {
+    const data = [
+      "subscriptionId",
+      "fxaUid",
+      "stripeEmail",
+      "isExcluded",
+      "amountRefunded",
+      "isOwed",
+      "error"
+    ];
+
+    const reportCSV = data.join(',') + '\n';
+
+    await writeFile(this.outputFile, reportCSV, {
+      encoding: 'utf-8',
+    });
+  }
+
+  async writeReport(args: {
+    subscription: Stripe.Subscription,
+    customer: Stripe.Customer | null,
+    isExcluded: boolean,
+    amountRefunded: number | null,
+    isOwed: boolean,
+    error: boolean
+  }) {
+    const data = [
+      args.subscription.id,
+      args.customer?.metadata.uid,
+      `"${args.customer?.email}"`,
+      String(args.isExcluded),
+      args.amountRefunded || "-",
```

Comment:
style nit: `-` could read as "negative number but we ran into an error". I'd prefer `--`, `0`, or an empty value here to avoid confusion